core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockPusher.scala

```diff
@@ -327,10 +338,32 @@ private[spark] class ShuffleBlockPusher(conf: SparkConf) extends Logging {
         s"stop.")
       return false
     } else {
+      if (remainingBlocks.isEmpty && pushRequests.isEmpty && deferredPushRequests.isEmpty) {
```

Comment:
It looks like the `remainingBlocks` is only for the current request. So it's possible that `remainingBlocks` could be non-empty for other ongoing requests (which are not either in `pushRequests` or `deferredPushRequests` ).

Reply:
Sorry, can you please explain this comment a bit more?

Reply:
@venkata91  Let me take an example:
Assuming we have request 1 and request 2 in total and each has 10 blocks. Then, we start request 1 and request 2 together. When we reach here, the empty`remainingBlocks` is only for request 1. And we'd see `pushRequests` and `deferredPushRequests` are both empty. However, request 2 is still ongoing and its remaining blocks could be not empty.

Reply:
`pushUpToMax` invokes `sendRequest` in multiple cases - each with a different `PushRequest`.
`remainingBlocks` is based on the current `PushRequest` - the remaining blocks for that push request.

Consider @Ngone51's example - suppose we have 2 total push requests - request1 and request2.
If we have request1 and request2 getting pushed concurrently (from `pushUpToMax`), completion of request1 does not mean all push has completed - request2 is still ongoing.

In this scenario, `pushRequests` and `deferredPushRequests` would be empty - since request1 and request2 were started together.
We will have `remainingBlocks` for request1 empty - but the concurrent request2 would not have completed.

See if we can leverage `reqsInFlight` instead.
Hopefully this clarifies.